pylabrobot/liquid_handling/backends/hamilton/STAR.py

```diff
@@ -385,10 +386,32 @@ def __init__(self, num_channels: int = 8, read_timeout: int = 5, **kwargs):
     self.dev: Optional[usb.core.Device] = None
     self._tip_types: dict[str, int] = {}
     self.num_channels = num_channels
+    self._iswap_parked: bool = False
 
     self.read_endpoint: Optional[usb.core.Endpoint] = None
     self.write_endpoint: Optional[usb.core.Endpoint] = None
 
+  @property
+  def iswap_parked(self)->bool:
+    return self._iswap_parked
+
+  def need_iswap_parked(method: Callable): # pylint: disable=no-self-argument
+    """Ensure that the iSWAP is in parked position before running command.
+
+    If the iSWAP is not parked, it get's parked before running the command.
+    """
+
+    @functools.wraps(method)
+    def wrapper(self, *args, **kwargs):
+      if self.iswap_parked:
+        result = method(self, *args, **kwargs) # pylint: disable=not-callable
+      else:
+        self.park_iswap()
+        result = method(self, *args, **kwargs) # pylint: disable=not-callable
```

Comment:
This can be further refactored as

```py
if not self.iswap_parked:
  self.park_iswap()

result = method(self, *args, **kwargs) # pylint: disable=not-callable
```

Reply:
addressed